prescient.el

```diff
@@ -358,7 +359,9 @@ enclose literal substrings with capture groups."
           (pcase method
             (`literal
              (prescient--with-group
-              (regexp-quote subquery)
+              (if (eq search-default-mode #'char-fold-to-regexp)
+                  (char-fold-to-regexp (regexp-quote subquery))
```

Comment:
Why does this get run through `regexp-quote` first?

Reply:
Since this is `literal` search, we first need to quote all 'regexp` symbols. If we reverse the order, char folding breaks since it is implemented via regexps.

Reply:
My question is why is there `regexp-quote` at all? Doesn't `char-fold-to-regexp` take an arbitrary string and return a regexp? Why do we need to do extra quoting?

Reply:
Ah sorry, I misunderstood you. Indeed, there is no need to quote regexp. And even the documentation mentions that `char-fold-to-regexp` quotes by itself :sweat_smile: Will update PR later :) 

```emacs-lsip
ELISP> (regexp-quote "[a]+")
"\\[a]\\+"
ELISP> (char-fold-to-regexp "[a]+")
"[[﹇［]\\(?:a[̀-̄̆-̨̣̥̊̌̏̑]\\|[aªà-åāăąǎǟǡǻȁȃȧᵃḁạảấầẩẫậắằẳẵặₐⓐａ𝐚𝑎𝒂𝒶𝓪𝔞𝕒𝖆𝖺𝗮𝘢𝙖𝚊]\\)[]﹈］][+⁺₊﬩﹢＋]"
ELISP> (string-match (regexp-quote "[a]+") "[a]+")
0 (#o0, #x0, ?\C-@)
ELISP> (string-match (char-fold-to-regexp "[a]+") "[a]+")
0 (#o0, #x0, ?\C-@)
```